### PR TITLE
Fix `->findBy()` for UUID object

### LINKED ISSUE #4704: [3.8.0-rc.1] Reverse relation with UUID via `findBy` not working
## Description

- Given I have two content types: `event` and `post`
- Where `post` has a field `relatedevent`, which stores an `event` UUID
- When getting `post => event` relation via `find('page://event-id')` works OK
- But, the other way around `event => post` via `findBy` doesn't work when just using `uuid()`
  - This fails: `$this->kirby()->collection('blog/posts')->findBy('relatedevent', $this->uuid())`
  - This works: `$this->kirby()->collection('blog/posts')->findBy('relatedevent', $this->uuid()->toString())`

**Expected behavior**  

- It should work with just `$this->uuid()`, without the need to cast to string.

## Your setup

**Kirby Version** : `3.8.0-rc.1`

### PR DESCRIPTION
**Pull request by distantnative**
Fixes #4704 (see for example)

### CODE REVIEW

File: src/Toolkit/Collection.php [hunk 1]
```diff
@@ -412,6 +412,11 @@ public function find(...$keys)
 	 */
 	public function findBy(string $attribute, $value)
 	{
+		// $value: cast object to string, e.g. UUID object
+		if (is_object($value) === true) {
```
Comment (bastianallgeier) [inline src/Toolkit/Collection.php]: As Lukas posted in Discord, this one would be less prone to unexpected regressions. 

```suggestion
		if (is_a($value, 'Kirby\Uuid\Uuid') === true) {
```

Comment (distantnative) [reply src/Toolkit/Collection.php]: But then we should do that in Cms\Collection. And we need to get used to `instanceof`, not `is_a`.

File: src/Cms/Collection.php [hunk 2]
```diff
@@ -123,6 +124,23 @@ public function append(...$args)
...
+	/**
+	 * Find a single element by an attribute and its value
+	 *
+	 * @param string $attribute
+	 * @param mixed $value
+	 * @return mixed|null
+	 */
+	public function findBy(string $attribute, $value)
+	{
+		// $value: cast UUID object to string
```
Comment (lukasbestle) [inline src/Cms/Collection.php]:
```suggestion
		// $value: cast UUID object to string to allow uses
		// like `$pages->findBy('related', $page->uuid())`
```

**Review by lukasbestle — COMMENTED** [review]
IMO there should be an additional unit test for this in `Cms\CollectionTest`. The test in `HasUuidsTest` is more an integration test.

**Comment by distantnative** [comment]
👍 can't approve myself

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**